File: src/backend/CareFirstClinic.API/CareFirstClinic.API/Models/MedicalRecord.cs
namespace CareFirstClinic.API.Models
{
    public class MedicalRecord
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        public Guid PatientId { get; set; }
        public Guid DoctorId { get; set; }
        public Guid AppointmentId { get; set; }

        public string Diagnosis { get; set; } = string.Empty;  // Chẩn đoán
        public string? Symptoms { get; set; }                   // Triệu chứng
        public string? Notes { get; set; }
        public DateTime? FollowUpDate { get; set; }         // Ngày tái khám

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        // Navigation
        public Patient? Patient { get; set; }
        public Doctor? Doctor { get; set; }
        public Appointment? Appointment { get; set; }
        public Prescription? Prescription { get; set; }
    }
}

File: src/frontend/src/types/medicalRecord.ts
export interface MedicalRecord {
  id: string;
  appointmentId: string;
  patientId: string;
  patientName: string;
  doctorId: string;
  doctorName: string;
  diagnosis: string;
  symptoms?: string;
  bloodPressure?: number;
  heartRate?: number;
  temperature?: number;
  weight?: number;
  height?: number;
  notes?: string;
  followUpDate?: string;
  hasPrescription: boolean;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateMedicalRecordDTO {
  appointmentId: string;
  diagnosis: string;
  symptoms?: string;
  bloodPressure?: number;
  heartRate?: number;
  temperature?: number;
  weight?: number;
  height?: number;
  notes?: string;
  followUpDate?: string;
}

export interface UpdateMedicalRecordDTO {
  diagnosis: string;
  symptoms?: string;
  bloodPressure?: number;
  heartRate?: number;
  temperature?: number;
  weight?: number;
  height?: number;
  notes?: string;
  followUpDate?: string;
}

export interface MedicalRecordQueryParams {
  patientId?: string;
  doctorId?: string;
  diagnosis?: string;
  hasFollowUp?: boolean;
  page?: number;
  pageSize?: number;
  sortBy?: string;
  sortDir?: string;
}

export interface PagedResult<T> {
  items: T[];
  totalCount: number;
  totalPages: number;
  currentPage: number;
  pageSize: number;
}